lib/client.js

```diff
@@ -600,6 +604,63 @@ Request.prototype.unset = function(field){
 };
 
 /**
+ * Set a filter for accepted status codes (that don't result in an error)
+ *
+ * Examples:
+ *
+ *      req.get('/')
+ *        .set('Accept', 'application/json')
+ *        .set('X-API-Key', 'foobar')
+ *        .acceptedStatus(['2xx', '4xx', 500])
+ *        .end(callback);
+ *
+ *      req.get('/')
+ *        .set({ Accept: 'application/json', 'X-API-Key': 'foobar' })
+ *        .acceptedStatus(function(status) { return status === 200; })
+ *        .end(callback);
+ *
+ * @param {Array|Function} field
+ * @return {Request} for chaining
+ * @api public
+ */
+
+Request.prototype.acceptedStatus = function(filter) {
+  if (filter instanceof Array) {
+    this.isAcceptedStatus = function(status) {
+      for (var i = 0; i < filter.length; i++) {
+        if (typeof filter[i] === 'string') {
+          var regexStr = [].map.call(filter[i].toLowerCase(), function(x) {
+            if (x === 'x') {
+              return '\\d';
+            }
+
+            return x;
+          }).join('');
+
+          var regex = new RegExp(regexStr);
+          if (regex.test(String(status))) {
+            return true;
+          }
+
+          continue;
+        }
+
+        if (filter[i] === status) {
+          return true;
+        }
+      }
+
+      return false;
+    };
+
+    return this;
+  }
+
+  this.isAcceptedStatus = filter;
```

Comment:
The idea here was to save it to the object property, and it's actually called later in the range check when the response is received, so I'm unsure what you're referring to by saving.